### PR TITLE
Add private registry support and use Rancher mirror images for Monitoring subcharts

### PR DESCRIPTION
**Pull request by aiyengar2**
#### Note to Reviewers

I double checked the existence of the Rancher images referenced on DockerHub after https://github.com/rancher/image-mirror/pull/24 was merged in.

#### What changes are included in this PR?

This PR includes changes to Rancher monitoring that need to be applied on the subcharts as well, namely:
1. Updating the default images of the monitoring subcharts to use the Rancher mirrors
1. Updating the monitoring subcharts to use private registry

Related Issue: https://github.com/rancher/rancher/issues/28441

### CODE REVIEW

**Comment by aiyengar2** [comment]
rancher/busybox still pending creation due to https://github.com/rancher/image-mirror/pull/25

File: scripts/generate-patch [hunk 1]
```diff
@@ -58,16 +58,41 @@ for f in packages/*; do
 				mkdir -p ${f}/charts-original
 				curl -sLf ${url} | tar xvzf - -C ${f}/charts-original --strip ${fields} ${subdirectory} > /dev/null 2>&1
 			fi
+			if [[ -f ${f}/charts/requirements.yaml ]]; then
+				i=0
+				while ! [[ -z $(yq r ${f}/charts/requirements.yaml "dependencies[${i}]") ]]; do
+					subchart=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].name" | head -n 1)
+					version=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].version" | head -n 1)
+					helm_repo=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].repository" | head -n 1)
```
Comment (brendarearden) [inline scripts/generate-patch:66]: is there ever a time when a repository is not listed on a dependency?

Comment (aiyengar2) [reply scripts/generate-patch:66]: @brendarearden this particular change is actually introduced in https://github.com/rancher/charts/pull/579, not part of this PR.

Comment (aiyengar2) [reply scripts/generate-patch:66]: > is there ever a time when a repository is not listed on a dependency?

Just double checked tho; these are guaranteed by Helm based on the [docs](https://helm.sh/docs/topics/charts/#chart-dependencies)

**Comment by aiyengar2** [comment]
@brendarearden since your review, I've expanded the scope of this PR to include changes necessary for https://github.com/rancher/rancher/issues/28495 since those will also be dependent on the subchart changes being merged and fixed version numbers being introduced for the rancher-monitoring chart to support the script changes. 

Only the last commit that has been added relates to these changes; nothing else has been changed since you last reviewed.

Re-requesting your review if you get a chance!

**Review by brendarearden — APPROVED** [review]
LGTM

**Comment by aiyengar2** [comment]
> @brendarearden since your review, I've expanded the scope of this PR to include changes necessary for [rancher/rancher#28495](https://github.com/rancher/rancher/issues/28495) since those will also be dependent on the subchart changes being merged and fixed version numbers being introduced for the rancher-monitoring chart to support the script changes.
> 
> Only the last commit that has been added relates to these changes; nothing else has been changed since you last reviewed.
> 
> Re-requesting your review if you get a chance!

Ignore the above comment; I reverted the changes introduced in this PR for https://github.com/rancher/rancher/issues/28495 and put it into a new PR https://github.com/rancher/charts/pull/611 since it seems like more complex changes might be required.

Also rebased based on current dev-v2.5 head; this PR should be ready for review / merge.

**Review by PennyScissors — APPROVED** [review]
lgtm

File: packages/rancher-monitoring/rancher-monitoring.patch [hunk 2]
```diff
@@ -283,79 +535,95 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/crd
...
 --- packages/rancher-monitoring/charts-original/requirements.yaml
 +++ packages/rancher-monitoring/charts/requirements.yaml
-@@ -14,3 +14,98 @@
-     version: "5.3.*"
+@@ -1,16 +1,111 @@
+ dependencies:
+ 
+   - name: kube-state-metrics
+-    version: "2.8.*"
++    version: "2.8.14"
```
Comment (deniseschannon) [inline packages/rancher-monitoring/rancher-monitoring.patch:543]: Should this be hard-coded to a specific version? Not sure if this has been discussed elsewhere, but curious.

File: packages/rancher-monitoring/rancher-monitoring.patch [hunk 3]
```diff
@@ -283,79 +535,95 @@ diff -x '*.tgz' -x '*.lock' -uNr packages/rancher-monitoring/charts-original/crd
...
 +  - name: prometheus-adapter
-+    version: 2.4.*
++    version: "2.4.0"
 +    repository: https://kubernetes-charts.storage.googleapis.com/
 +    condition: prometheus-adapter.enabled
 +
 +  - name: rancher-pushprox
 +    alias: rkeControllerManager
-+    version: 0.1.0
++    version: "0.1.0"
```
Comment (deniseschannon) [inline packages/rancher-monitoring/rancher-monitoring.patch:566]: Can these formatting changes be avoided in the patch?

Comment (aiyengar2) [reply packages/rancher-monitoring/rancher-monitoring.patch:566]: yeah sure, will push a commit out

**Review by deniseschannon — APPROVED** [review]

**Pull request merged**